integration_tests/src/main/python/conftest.py

```diff
@@ -54,13 +54,17 @@ def array_columns_to_sort_locally():
 
 _allow_any_non_gpu = False
 _non_gpu_allowed = []
+_per_test_ansi_mode = None
 
 def is_allowing_any_non_gpu():
     return _allow_any_non_gpu
 
 def get_non_gpu_allowed():
     return _non_gpu_allowed
 
+def get_per_test_ansi_mode():
```

Comment:
I wonder if the logic problem could have been avoided if the names of the variable and function were better.  Note that the annotation is a "negative" (`disable_ansi_mode`).

```suggestion
def is_per_test_ansi_mode_enabled():
```
Likewise, the variable should probably be named `_per_test_ansi_mode_enabled`.